akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -843,10 +850,10 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
       // Don't mark gossip state as seen while exiting is in progress, e.g.
       // shutting down singleton actors. This delays removal of the member until
       // the exiting tasks have been completed.
-      if (exitingTasksInProgress)
-        latestGossip = winningGossip
-      else
-        latestGossip = winningGossip seen selfUniqueAddress
+      latestGossip =
+        (if (exitingTasksInProgress) winningGossip
+        else winningGossip seen selfUniqueAddress)
+          .pruneTombstones(System.currentTimeMillis() - PruneGossipTombstonesAfter.toMillis)
```

Comment:
Yeah, unsure, I "just put it somewhere", you thinking that it is too much overhead doing it every time here or that it just doesn't make sense?